index.c

```diff
@@ -2822,13 +2822,21 @@ int mutt_index_menu(struct MuttWindow *dlg)
         struct EmailList el = STAILQ_HEAD_INITIALIZER(el);
         el_add_tagged(&el, Context, cur.e, tag);
 
-        const bool delete_original =
-            (op == OP_SAVE) || (op == OP_DECODE_SAVE) || (op == OP_DECRYPT_SAVE);
-        const bool decode = (op == OP_DECODE_SAVE) || (op == OP_DECODE_COPY);
-        const bool decrypt = (op == OP_DECRYPT_SAVE) || (op == OP_DECRYPT_COPY);
+        const enum MessageSaveOpt save_opt =
+            ((op == OP_SAVE) || (op == OP_DECODE_SAVE) || (op == OP_DECRYPT_SAVE)) ?
+                SAVE_MOVE :
+                SAVE_COPY;
+
+        enum MessageTransformOpt transform_opt;
+        if ((op == OP_DECODE_SAVE) || (op == OP_DECODE_COPY))
+          transform_opt = TRANSFORM_DECODE;
+        else if ((op == OP_DECRYPT_SAVE) || (op == OP_DECRYPT_COPY))
+          transform_opt = TRANSFORM_DECRYPT;
+        else
+          transform_opt = TRANSFORM_NONE;
```

Comment:
```
enum MessageTransformOpt transform_opt =
    ((op == OP_DECODE_SAVE) || (op == OP_DECODE_COPY))
    ? TRANSFORM_DECODE
    : ((op == OP_DECRYPT_SAVE) || (op == OP_DECRYPT_COPY))
      ? TRANSFORM_DECRYPT
      : TRANSFORM_NONE;
```

not sure flatcap agrees, though

Reply:
I considered this initially, but I was pretty sure @flatcap won't like nested ternaries...
I personally like this, looks more readable to me.

Reply:
After clang-formatting, it looks like this:
```c
enum MessageTransformOpt transform_opt =
    ((op == OP_DECODE_SAVE) || (op == OP_DECODE_COPY))   ? TRANSFORM_DECODE :
    ((op == OP_DECRYPT_SAVE) || (op == OP_DECRYPT_COPY)) ? TRANSFORM_DECRYPT :
                                                           TRANSFORM_NONE;
```

which **is** quite legible, but that's because the constants coincide with the line length.
Too often, that's not the case.